src/main/java/org/junit/internal/Throwables.java

```diff
@@ -104,8 +105,31 @@ public static String getTrimmedStackTrace(Throwable exception) {
         return Collections.emptyList();
     }
 
+    private static final Method getSuppressed = initGetSuppressed();
+
+    private static Method initGetSuppressed() {
+        try {
+            return Throwable.class.getMethod("getSuppressed");
+        } catch (Throwable e) {
+            return null;
+        }
+    }
+
+    private static boolean hasSuppressed(Throwable exception) {
+        if (getSuppressed != null) {
```

Comment:
nit: I personally prefer existing out on error conditions early, to avoid over-intentation:

    if (getSuppressed == null) {
      return false;
    }
    try {
      ...
    } catch (Throwable e) {
      return false;
    }